### PR TITLE
Add charset option to text, html part

### LINKED ISSUE #119: Can't pass charset to Content-Type headers
Omitting the `charset` in the `Content-Type` header makes some email clients use arbitrary encodings, for example Thunderbird seems to default to ISO-8859-1/15. I would like some input on what the API should look like to allow passing this, and to have it default to `utf-8`.

I'm thinking of adding a `charset` field to the `Mail.Message` struct and change `build_html/build_text` to also allow passing an optional `charset` param. 

I can work on a PR once we decide if this is the right approach.

Thanks

### PR DESCRIPTION
**Pull request by marocchino**
Closes #119 

## Changes proposed in this pull request

- add `:charset` option to put_text, put_html, build_text, build_html
- can receive content type with charset in get_html

### CODE REVIEW

File: lib/mail.ex [hunk 1]
```diff
@@ -67,8 +82,12 @@ defmodule Mail do
     end)
   end
 
-  def get_text(%Mail.Message{headers: %{"content-type" => "text/plain" <> _}} = message), do: message
-  def get_text(%Mail.Message{headers: %{"content-type" => ["text/plain" | _]}} = message), do: message
+  def get_text(%Mail.Message{headers: %{"content-type" => "text/plain" <> _}} = message),
+    do: message
+
+  def get_text(%Mail.Message{headers: %{"content-type" => ["text/plain" | _]}} = message),
```
Comment (marocchino) [inline lib/mail.ex:88]: Not a part of change. It was auto fixed by lint. 😅

File: lib/mail.ex [hunk 2]
```diff
@@ -78,21 +97,36 @@ defmodule Mail do
...
+          ["text/html", {"charset", charset}]
+
+        _else ->
+          "text/html"
+      end
+
     Mail.Message.put_body(message, body)
     |> Mail.Message.put_header(:content_transfer_encoding, :quoted_printable)
-    |> Mail.Message.put_content_type("text/html")
+    |> Mail.Message.put_content_type(content_type)
```
Comment (bcardarella) [inline lib/mail.ex:129]: If this is meant to be specifically a "html" part why would we want the `content_type` to be overridable?

File: lib/mail.ex [hunk 3]
```diff
@@ -34,21 +34,36 @@ defmodule Mail do
...
+          ["text/plain", {"charset", charset}]
+
+        _else ->
+          "text/plain"
+      end
+
     Mail.Message.put_body(message, body)
     |> Mail.Message.put_header(:content_transfer_encoding, :quoted_printable)
-    |> Mail.Message.put_content_type("text/plain")
+    |> Mail.Message.put_content_type(content_type)
```
Comment (bcardarella) [inline lib/mail.ex:66]: If this is meant to be a text part why would we want the `content_type` to be overridable?

File: lib/mail.ex (hunk 2 above)
Comment (bcardarella) [reply lib/mail.ex:129]: Oh I see below, it looks like you force the content type but allow charset to be configurable?

Comment (marocchino) [reply lib/mail.ex:129]: Correct. Some local mail clients do not display mail properly if charset is not set. (Japanese in my case)

**Pull request merged**